src/react-hook/react-hook.tsx

```diff
@@ -0,0 +1,31 @@
+import React from "react";
+import * as Config from "../module-config/module-config";
+
+export const ModuleNameContext = React.createContext(null);
+
+// As written, this will not work to provide config to more than one module.
+// `config` will be shared across client modules, resulting in all modules
+// receiving the configuration for the module that happens to call this first.
+//
+// When using useState to manage the config, the browser tab goes into infinite
+// recursion via setConfig().
+//
```

Comment:
This comment sums up where I'm at.

>// As written, this will not work to provide config to more than one module.
// `config` will be shared across client modules, resulting in all modules
// receiving the configuration for the module that happens to call this first.
//
// When using useState to manage the config, the browser tab goes into infinite
// recursion via setConfig().